build_web_compilers/README.md

```diff
@@ -120,16 +265,13 @@ targets:
       build_web_compilers:entrypoint:
         options:
           compiler: dart2wasm
+          # List flags that should be forwarded to `dart compile wasm`
           dart2wasm_args:
-          - -DSOME_VAR=some value
-          - -DANOTHER_VAR=true
+          - -O2
 ```
 
-These may also be specified on the command line with a `--define` argument.
-
-```sh
-webdev serve -- '--define=build_web_compilers:ddc=environment={"SOME_VAR":"changed"}'
-```
+Note that the `compiler` option takes precedence over the new `compilers` and
```

Comment:
We should probably make this an error during option parsing

Reply:
The readme is outdated now, I've changed it so that `compilers` takes precedence.

This is a bit unfortunate, but we can't make this an error because the keys from the user's `build.yaml` are merged with the defaults in `build_web_compilers` - so we can't assume that something's off just because both are set.
The current approach is to use the existing `compiler` option in the defaults (so that `compilers` never appears in the map unless explicitly enabled) and then have `compilers` take precedence.

I wonder if we should unify these into a single option, where `compiler` is either a string (old) or a map (new)? That makes it harder to be in an ambiguous state.

Reply:
I think how you have it is fine. The plural reads nicer now that multiple can be configured, and it is a cleaner break.